Marlin/src/gcode/calibrate/M666.cpp

```diff
@@ -94,7 +94,13 @@
       #endif
       #if ENABLED(Z_MULTI_ENDSTOPS)
         #define _ECHO_ZADJ(N) SERIAL_ECHOPAIR(" Z" STRINGIFY(N) ":", endstops.z##N##_endstop_adj);
-        REPEAT_S(2, INCREMENT(NUM_Z_STEPPER_DRIVERS), _ECHO_ZADJ)
+        #if NUM_Z_STEPPER_DRIVERS >= 3
+          _ECHO_ZADJ(2)
```

Comment:
Surely when there are only two Z endstops, we want to be able to adjust the second one relative to the first. Is there some reason this has to start with 3, or is this just emulating some other instance where this logic is being applied?
```cpp
    #if ENABLED(Z_MULTI_ENDSTOPS)
      static float z2_endstop_adj;
    #endif
    #if ENABLED(Z_MULTI_ENDSTOPS) && NUM_Z_STEPPER_DRIVERS >= 3
      static float z3_endstop_adj;
    #endif
    #if ENABLED(Z_MULTI_ENDSTOPS) && NUM_Z_STEPPER_DRIVERS >= 4
      static float z4_endstop_adj;
    #endif
```